app/components/icon_component.html.erb

```diff
@@ -1,10 +1,4 @@
-<%= content_tag(
-      :svg,
-      aria: { hidden: true },
-      focusable: 'false',
-      role: 'img',
-      **tag_options,
-      class: css_class,
-    ) do %>
-  <use href="<%= icon_path %>"></use>
+<%= content_tag(:span, '', **tag_options, id: "icon-#{unique_id}", class: css_class) %>
+<%= content_tag(:style, nonce: content_security_policy_nonce) do %>
+  <%= "#icon-#{unique_id} { mask-image: url(#{icon_path}); -webkit-mask-image: url(#{icon_path}); mask-size: 100%; -webkit-mask-size: 100%; background-color: currentColor; }" %>
```

Comment:
What if we used ActiveSupport's `String#squish` to let us author it with line breaks for readability but remove them for serialization?

```suggestion
  <%= <<-STR.squish
    #icon-#{unique_id} {
      mask-image: url(#{icon_path});
      -webkit-mask-image: url(#{icon_path});
      mask-size: 100%;
      -webkit-mask-size: 100%;
      background-color: currentColor;
    }
    STR
    %>
```

Reply:
This motivated me toward #10067, which I'm hoping can fix the issue with component stylesheets so we can move the `mask-size` and `background-color` properties out and shorten it to something more like this, which hopefully reduces the readability concern a bit?

```suggestion
  <%= "#icon-#{unique_id} { mask-image: url(#{icon_path}); -webkit-mask-image: url(#{icon_path}); }" %>
```

Reply:
I think it still might be nice with some line breaks, but that is a little clearer